packages/runtime/container-runtime/src/test/batching.spec.ts

```diff
@@ -0,0 +1,229 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { AttachState, ICriticalContainerError } from "@fluidframework/container-definitions";
+import { IContainerContext } from "@fluidframework/container-definitions/internal";
+import { FluidErrorTypes } from "@fluidframework/core-interfaces/internal";
+import {
+	MessageType,
+	ISequencedDocumentMessage,
+} from "@fluidframework/driver-definitions/internal";
+import { MockLogger, mixinMonitoringContext } from "@fluidframework/telemetry-utils/internal";
+import {
+	MockAudience,
+	MockDeltaManager,
+	MockQuorumClients,
+	validateAssertionError,
+} from "@fluidframework/test-runtime-utils/internal";
+
+import type { ChannelCollection } from "../channelCollection.js";
+import { ContainerRuntime } from "../containerRuntime.js";
+import { ContainerMessageType } from "../messageTypes.js";
+
+describe("Runtime batching", () => {
+	const mockClientId = "mockClientId";
+
+	const getMockContext = (deltaManager: MockDeltaManager): Partial<IContainerContext> => {
+		const mockContext = {
+			attachState: AttachState.Attached,
+			deltaManager,
+			audience: new MockAudience(),
+			quorum: new MockQuorumClients(),
+			taggedLogger: mixinMonitoringContext(new MockLogger()).logger,
+			clientDetails: { capabilities: { interactive: true } },
+			closeFn: (_error?: ICriticalContainerError): void => {},
+			updateDirtyContainerState: (_dirty: boolean) => {},
+			getLoadedFromVersion: () => undefined,
+			clientId: mockClientId,
+			connected: true,
+		};
+		return mockContext;
+	};
+
+	const mockProvideEntryPoint = async () => ({
+		myProp: "myValue",
+	});
+
+	let containerRuntime: ContainerRuntime;
+	let mockDeltaManager: MockDeltaManager;
+
+	beforeEach(async () => {
+		mockDeltaManager = new MockDeltaManager();
+		containerRuntime = await ContainerRuntime.loadRuntime({
+			context: getMockContext(mockDeltaManager) as IContainerContext,
+			registryEntries: [],
+			existing: false,
+			runtimeOptions: {},
+			provideEntryPoint: mockProvideEntryPoint,
+		});
+	});
+
+	/** Overwrites channelCollection property to make process a no-op */
```

Comment:
I think Sinon can do this in about 1 line of code but would have to look up what it is.